optuna/integration/pytorch_distributed.py

```diff
@@ -19,11 +23,54 @@
     import torch.distributed as dist
 
 
+_T = TypeVar("_T")
+_P = ParamSpec("_P")
+
+
 _suggest_deprecated_msg = (
     "Use :func:`~optuna.integration.TorchDistributedTrial.suggest_float` instead."
 )
 
 
+def updates_properties(f: Callable[_P, _T]) -> Callable[_P, _T]:
+    """Method decorator to fetch updated trial properties from rank 0 after ``f`` is run.
+
+    This decorator ensures trial properties (params, distributions, etc.) on all distributed
+    processes are up-to-date with the wrapped trial stored on rank 0.
+    It should be applied to all :class:`~optuna.integration.TorchDistributedTrial`
+    methods that update property values.
+    """
+
+    @functools.wraps(f)
+    def wrapped(*args: _P.args, **kwargs: _P.kwargs) -> _T:
+        self: TorchDistributedTrial = args[0]
```

Comment:
Thank you for your update. Could you ignore the type checking tentitatively?
```suggestion
        # TODO(nlgranger): Remove type ignore after mypy includes
        # https://github.com/python/mypy/pull/12668
        self: TorchDistributedTrial = args[0]  # type: ignore
```